### PR TITLE
ISPN-8719 KeySet.(iterator|spliterator|stream) not compatible with

### PR DESCRIPTION
**Pull request by wburns**
versions before 9.1

* Added iterator methods to Codecs
* Put appropriate methods on each Codec version

https://issues.jboss.org/browse/ISPN-8719

### CODE REVIEW

**Comment by tristantarrant** [comment]
[ERROR] /home/infinispan/workspace/Infinispan_PR-5903-FZGGAVW2K7EIQ7VCW4FTI23ZZSG34WXQMQXSAMW3BO4EKBUMA6KQ/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/IterationStartOperation.java:3:15: Unused import - java.util.Arrays.stream. [UnusedImports]
[ERROR] /home/infinispan/workspace/Infinispan_PR-5903-FZGGAVW2K7EIQ7VCW4FTI23ZZSG34WXQMQXSAMW3BO4EKBUMA6KQ/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/IterationStartOperation.java:6:8: Unused import - java.util.BitSet. [UnusedImports]
[ERROR] /home/infinispan/workspace/Infinispan_PR-5903-FZGGAVW2K7EIQ7VCW4FTI23ZZSG34WXQMQXSAMW3BO4EKBUMA6KQ/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/IterationStartOperation.java:50: White spaces at the end of line [RegexpSingleline]

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/Codec.java [hunk 1]
```diff
@@ -74,4 +78,49 @@ void writeClientListenerParams(Transport transport, ClientListener clientListene
...
+   default int readProjectionSize(Transport transport) {
+      return 0;
+   }
+
+   /**
+    * Iteration read to tell if metadata is present for entry
+    * @param transport
+    * @return
+    */
+   default short readMeta(Transport transport) {
```
Comment (gustavonalle) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/Codec.java]: Rename it to ```readMetadata```?

Comment (gustavonalle) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/Codec.java]: Nevermind, just realized this is a backport

**Comment by wburns** [comment]
Should be fixed.

**Pull request merged**